src/integrity-checker.js

```diff
@@ -33,30 +39,37 @@ export default class InstallationIntegrityChecker {
    * write a new one.
    */
 
-  async _getIntegrityHashLocation(): Promise<string> {
+  async _getIntegrityHashLocation(usedRegistries?: Set<RegistryNames>): Promise<IntegrityHashLocation> {
     // build up possible folders
+    let registries = registryNames;
+    if (usedRegistries && usedRegistries.size > 0) {
+      registries = usedRegistries;
+    }
```

Comment:
Removed my previous comment.
Actually it is fine to pass falsy usedRegistries when generating the integrity file, e.g. running `yarn install` for empty package.json.
Also Yarn won't generate multiple integrity files because this method checks whether one exists already and reuses that location